let Joi,User,CustomErrorHandler;_e02‍.x([["default",()=>_e02‍.o]]);_e02‍.w("joi",[["default",["Joi"],function(v){Joi=v}]]);_e02‍.w("../models",[["User",["User"],function(v){User=v}]]);_e02‍.w("../services",[["CustomErrorHandler",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);


const signUpDataValidate=async(req,res,next)=>{
    const signupSchema=Joi.object({
        name:Joi.string().min(5).max(50).required(),
        username:Joi.string(),
        email:Joi.string().email(),
        password:Joi.string()
        .pattern(new RegExp('^[a-zA-Z0-9]{3,30}$')).required(),
        repeatPassword:Joi.ref('password'),
        bio:Joi.string().required()
    });
    const{error}=signupSchema.validate(req.body);
    if(error){
        return next(error);
    }
    const exists=await User.exists({username:req.body.username,email:req.body.email});
    if(exists){
        return next(CustomErrorHandler.alreadyExists('This email or username is already taken.'));
    }
    next();
}
_e02‍.d(signUpDataValidate); 


